### DISCUSSION TITLE
What happens when adding Powerwalls?

### DISCUSSION THREAD

**Discussion by clukas1967** [jasonacox/Powerwall-Dashboard#241] [Q&A]
Hi.  Have Tesla PV system with one PW live since 2019.  Running PW Dashboard (PWD) for 9 days now and have pulled all my old history.

I got in queue a year ago to add 2 additional PWs. They are finally shipping and permitting is under way.  Should be live sometime next month.

My questions are:

- What happens to my current PWD display when this happens?   Does it appear to PWD that my PW just tripled in size?  From looking at some other users' screen shots it seems like this is what happens but want to be sure.
- Do I need to do anything pre-emptively in InfluxDB or any other system part to protect against any corruption?   Due to say the meaning of relative metrics changing?
- How is the "charge %" stored natively in influxDB?  Does this come as a % directly from the Gateway, or is it imputed somewhere in the Telegraf/Influx/Grafana pipeline?
- For example, the "80%" mark of PW charge level today will mean something completely different two months from now.  Will the Telegraf/Influx/Grafana pipeline correctly display overall system charge state before and after the PW expansion?   (e.g. 80% of 1 PW prior to new install vs. 80% of multiple PWs after install).   I could see that if it's the Gateway doing all the work in this respect that the actual system size details would be transparent to PW Dashboard.
- Does anybody with multiple PWs installed track them separately in PW Dashboard?  If so can you point me to a discussion thread and widgets?   

Thanks for all the great work.

**Comment by jasonacox** [comment]
Hi @clukas1967 !   Congrats on getting the additional Powerwalls!  In hindsight, I would have added more panels and more Powerwalls.  😉 

> What happens to my current PWD display when this happens? Does it appear to PWD that my PW just tripled in size? From looking at some other users' screen shots it seems like this is what happens but want to be sure.

Yes, it will just show up as triple in size. If you have a Powerwall+ it will show up in the capacity graph as two new Powerwalls. The graphs are pre-configured for up to 6 Powerwalls but will not show if the data is null (missing).

> Do I need to do anything pre-emptively in InfluxDB or any other system part to protect against any corruption? Due to say the meaning of relative metrics changing?

No, it should just start working.  I create backups of my InfluxDB directory on a regular basis anyway (see backup.sh).

> How is the "charge %" stored natively in influxDB? Does this come as a % directly from the Gateway, or is it imputed somewhere in the Telegraf/Influx/Grafana pipeline?

The charge % comes from the Tesla Gateway so it is doing the calculation. You can see it via http://localhost:8675/soe (where localhost is your PWD host). The graphs only show this value and don't care about how many Powerwalls you have.

> Does anybody with multiple PWs installed track them separately in PW Dashboard? If so can you point me to a discussion thread and widgets?

The Powerwall+ section has these graphs (I have two powerwalls). 

<img width="569" alt="image" src="https://user-images.githubusercontent.com/836718/231354773-8911e085-13ac-4be1-84bb-f3ad6ef66afc.png">
<img width="622" alt="image" src="https://user-images.githubusercontent.com/836718/231354839-3a8b0c1a-a449-4a83-b9ce-8f196fbc8ecc.png">
